specutils/io/default_loaders/sdss_v.py

```diff
@@ -578,28 +441,71 @@ def _load_mwmVisit_or_mwmStar_hdu(hdulist: HDUList, hdu: int, **kwargs):
     meta["snr"] = np.array(hdulist[hdu].data["snr"])
 
     # Add identifiers (obj, telescope, mjd, datatype)
-    # TODO: need to see what metadata we're interested in for the MWM files.
     meta["telescope"] = hdulist[hdu].data["telescope"]
     meta["instrument"] = hdulist[hdu].header.get("INSTRMNT")
-    try:
+    try:  # get obj if exists
         meta["obj"] = hdulist[hdu].data["obj"]
     except KeyError:
         pass
+
+    # choose between mwmVisit/Star via KeyError except
     try:
-        meta["date"] = hdulist[hdu].data["date_obs"]
-        meta["mjd"] = hdulist[hdu].data["mjd"]
+        meta['mjd'] = hdulist[hdu].data['mjd']
         meta["datatype"] = "mwmVisit"
     except KeyError:
-        meta["mjd"] = (str(hdulist[hdu].data["min_mjd"][0]) + "->" +
-                       str(hdulist[hdu].data["max_mjd"][0]))
+        meta["min_mjd"] = str(hdulist[hdu].data["min_mjd"][0])
+        meta["max_mjd"] = str(hdulist[hdu].data["max_mjd"][0])
         meta["datatype"] = "mwmStar"
     finally:
         meta["name"] = hdulist[hdu].name
+        meta["sdss_id"] = hdulist[hdu].data['sdss_id']
 
-    return Spectrum1D(
-        spectral_axis=spectral_axis,
-        flux=flux,
-        uncertainty=e_flux,
-        mask=mask,
-        meta=meta,
-    )
+    # drop back a list of Spectrum1Ds to unpack
+    metadicts = _split_mwm_meta_dict(meta)
+    return [
+        Spectrum1D(
+            spectral_axis=spectral_axis,
+            flux=flux[i],
+            uncertainty=e_flux[i],
+            mask=mask[i],
+            meta=metadicts[i],
+        ) for i in range(flux.shape[0])
+    ]
+
```

Comment:
we will need to do a similar thing for the `SDSS-V spec` loader, but only for any of the spec-full files.

Reply:
Aren't `spec-full` visits dispersed across the HDUs past the `spall` and `zall` HDU's? Or am I mistaken about the format?

Reply:
Yeah that's right.  They're appended at the end as 
```
5  MJD_EXP_60052-00    1 BinTableHDU    273   4648R x 8C   [E, E, E, J, J, E, E, E]   
6  MJD_EXP_60052-01    1 BinTableHDU    271   4648R x 8C   [E, E, E, J, J, E, E, E]   
7  MJD_EXP_60052-02    1 BinTableHDU    273   4648R x 8C   [E, E, E, J, J, E, E, E]   
8  MJD_EXP_60052-03    1 BinTableHDU    273   4648R x 8C   [E, E, E, J, J, E, E, E] 
...  
```
So, yeah, a change isn't needed here since they are 1d spectra in separate extensions.